<launch>
    <!-- dataset path args -->
    <arg name="setting_path" doc="Path of SLAM and camera config .yaml file"
         default="$(find orb_slam2_dense)/params/RTAB_DATASET.yaml" />
    <arg name="dataset_path" doc="Path of rosbag dataset, if `use_sim is enabled`"
         default="$(find orb_slam2_dense)/Vocabulary/2022-09-23-22-29-40.bag" />

     <!-- use node flag args -->
    <arg name="use_sim" default="true"/>
    <arg name="use_rviz" default="true" /> <!--true-->
    <!-- broadcast frame id args -->
    <arg name="map_frame" default="map" />
    <arg name="odom_frame" default="odom" />
    <arg name="footprint_frame" default="base_link_" />
    <arg name="optical_frame" default="camera_link_" />


     <!-- rosbag node-->
    <group if="$(arg use_sim)" >\
        <node pkg="rosbag" type="play" name="playbag" args="$(arg dataset_path)"/>
    </group>
     
    
    

     <group if="$(arg use_rviz)" >
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find orb_slam2_dense)/rviz/rgbd.rviz"  output="screen"/>
    </group>

    <!-- ORB-SLAM2_DENSE -->
    <include file="$(find orb_slam2_dense)/launch/rgbd_launch.launch" >
        <arg name="setting_path" value="$(arg setting_path)" />
        <arg name="use_rviz" value="$(arg use_rviz)" />
        <arg name="map_frame" value="$(arg map_frame)" />
        <arg name="odom_frame" value="$(arg odom_frame)" />
        <arg name="footprint_frame" value="$(arg footprint_frame)" />
        <arg name="optical_frame" value="$(arg optical_frame)" />
    </include>

    <!-- octomap server
    <node pkg="octomap_server" type="octomap_server_node" name="octomap_server_node" output="screen">
        <remap from="cloud_in" to="/map_cloud" />
        <param name="frame_id" value="$(arg map_frame)" />
        <param name="base_frame_id" value="$(arg footprint_frame)" />
        <param name="pointcloud_min_z" value="0.2" />
        <param name="pointcloud_max_z" value="3" />
    </node>
    -->


    <!-- static tf publisher -->
    <!-- 0 0 0.6 -1.5708 0 -1.705 0.326921-->
    <node pkg="tf" type="static_transform_publisher" name="footprint2optical"
          args="0 0 0.326921 -1.5708 0 -1.5708 $(arg footprint_frame) $(arg optical_frame) 30"  output="screen"/>
          
          
    <node name="image_transport_rgb" pkg="image_transport" type="republish" 
          args="compressed in:=/camera/data_throttled_image raw out:=/camera/rgb/image_color" output="screen">
	</node>
	
	
	<node name="image_transport_d" pkg="image_transport" type="republish"
	      args="compressedDepth in:=/camera/data_throttled_image_depth raw out:=/camera/depth/image" output="screen">
	</node>
   
    
    
</launch>
